### DISCUSSION TITLE
Primal heuristics

### DISCUSSION THREAD

**Discussion by th-yoon** [ds4dm/ecole#178] [Questions] [answered]
Does ecole support generating primal heuristic solutions from SCIP?

**Comment by gasse** [comment, marked as answer]
Coming soon :)

Edit: a bit more info, we are currently working on a new environment for learning to generate primal solutions. Developments are almost complete at this point, you can expect for it to be released in a few weeks.

**Reply by th-yoon** [reply]
Thanks for your reply. Then would it be possible to retrieve the internal feasible solutions from SCIP with current version of ecole?

**Reply by AntoinePrv** [reply]
You can use all the `pyscipopt` method on the `Model` using `env.model.as_pyscipopt()`. Not sure exactly how you get the solutions in PySCIPOpt though, but it should be feasible.

**Reply by gasse** [reply]
Yes, you have a few methods there that should do what you want:

https://github.com/scipopt/PySCIPOpt/blob/master/src/pyscipopt/scip.pyx#L4134

https://github.com/scipopt/PySCIPOpt/blob/master/src/pyscipopt/scip.pyx#L4147

https://github.com/scipopt/PySCIPOpt/blob/master/src/pyscipopt/scip.pyx#L4152

**Reply by th-yoon** [reply]
Awesome thanks!
